src/nisarqa/utils/plotting.py

```diff
@@ -3534,15 +3534,44 @@ def plot_connected_components_layer(
     # Hint: Visually, the "boundary" between two integer labels is where the
     # colorbar changes from one color to the next color.
 
+    # Kludge: the `labels` returned above use the same dtype as `cc_full`.
+    # As of March 2024, the product spec for the CC layer specifies a
+    # dtype of unsigned int16 with a fill value of 65535. For the CC layer,
+    # `0` designates pixels with invalid unwrapping, so `labels` will
+    # typically be [0, ..., 65535].
+    # Because 0 and 65535 are the min and max values for uint32, when
+    # we try to subtract 1 and add 1 to those values below when creating the
+    # `boundaries` variable, we'll run into overflow errors.
+    # Prior to NumPy 2.0 (e.g. NumPy 1.26), the datatype would automatically
+    # be promoted to avoid overflow. Starting with NumPy 2.0, datatypes were
+    # no longer promoted. So, manually promote the datatype to avoid overflow.
+    # Sources:
+    #     [1] https://numpy.org/doc/stable/numpy_2_0_migration_guide.html#changes-to-numpy-data-type-promotion
+    #     [2] https://numpy.org/neps/nep-0050-scalar-promotion.html
+    if (np.nanmin(labels) <= (-(2**63)) + 1) or (
+        np.nanmax(labels) >= (2**63 - 1) - 1
+    ):
```

Comment:
Let's use [`np.iinfo()`](https://numpy.org/doc/2.2/reference/generated/numpy.iinfo.html) here

```suggestion
    info = np.iinfo(np.int64)
    if (np.min(labels) <= info.min) or (np.max(labels) >= info.max):
```